presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastoreClient.java

```diff
@@ -44,18 +46,28 @@
 import java.util.List;
 import java.util.Map;
 
+import static com.google.common.reflect.Reflection.newProxy;
 import static java.util.Objects.requireNonNull;
 
 public class ThriftHiveMetastoreClient
         implements ThriftMetastoreClient
 {
+    private static final Logger log = Logger.get(ThriftHiveMetastoreClient.class);
+
     private final TTransport transport;
-    private final ThriftHiveMetastore.Client client;
+    private final ThriftHiveMetastore.Iface client;
 
     public ThriftHiveMetastoreClient(TTransport transport)
     {
         this.transport = requireNonNull(transport, "transport is null");
-        this.client = new ThriftHiveMetastore.Client(new TBinaryProtocol(transport));
+        ThriftHiveMetastore.Client client = new ThriftHiveMetastore.Client(new TBinaryProtocol(transport));
+        if (log.isDebugEnabled()) {
+            // TODO compile ThriftHiveMetastore.Iface with `-parameters` to provide meaningful argument names when logging.
```

Comment:
We can actually get the parameter names using the [parameter names](https://github.com/airlift/parameternames) library from the `Client` class. It reads the parameter names from the bytecode debug symbols (but they don't exist for interfaces since interfaces don't have code).

We could have the invocation handler take a class instance, and it could build a map at startup of all the methods in the interface using the names from the class.

Reply:
Super cool!